lib/hammer_cli_foreman/attribute.rb

```diff
@@ -0,0 +1,282 @@
+      params['compute_attribute'] = profile['compute_attributes'].select { |hash| hash['compute_resource_id'] == options['option_compute_resource_id']}[0] || {}
+      params
+    end
+
+    class Create < HammerCLIForeman::CreateCommand
+        desc _('Create compute profile set of values.')
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+
+      extend_help do |h|
+        ::HammerCLIForeman::ComputeResources.extend_help(h, :all)
+      end
+
+      validate_options do
+        any(:option_compute_profile_id, :option_compute_profile_name ).required
+        any(:option_compute_resource_id, :option_compute_resource_name).required
+      end
+
+      def request_params
+        params = super
+        params['compute_attribute']['vm_attrs'] = option_compute_attributes || {}
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = {new_element_id => option_interface}
+        end
+        if option_volume
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = {new_element_id => option_volume}
+        end
+        params
+      end
+
+      success_message _('Compute profile attributes are set.')
+      failure_message _('Could not set the compute profile attributes')
+      build_options
+    end
+
+    class Update < HammerCLIForeman::UpdateCommand
+      desc _('Update compute profile values.')
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+
+      extend_help do |h|
+        ::HammerCLIForeman::ComputeResources.extend_help(h, :all)
+      end
+
+      validate_options do
+        any(:option_compute_profile_id, :option_compute_profile_name ).required
+        any(:option_compute_resource_id, :option_compute_resource_name).required
+        any(:option_interface, :option_volume, :option_compute_attributes).required
```

Comment:
This validation won't work. Our validators do not work well with multivalued options. We either need to fix the validator or remove this validation.